datamapplot/interactive_rendering.py

```diff
@@ -875,6 +946,72 @@ def label_text_and_polygon_dataframes(
                     )
                 ]
             )
+        if include_related_points:
+            related_points.append(np.where(cluster_mask))
+            points_bounds.append(compute_percentile_bounds(cluster_points))
+        if parents is not None:
+            if len(parents[0]):
+                # Get the provenance (cluster membership at different heirarchical layers).
+                # This should be consistent.(??)
+                p = list(
+                    np.median(parents[0][:, cluster_mask], axis=1)
+                    .astype(int)
+                    .astype(str)
+                ) + [str(i)]
+                label_ids.append("_".join(p))
+                parent_ids.append("_".join(p[:-1]))
+            else:
+                label_ids.append(str(i))
+                parent_ids.append(None)
+
+    if parents is not None:
+        # do the same for unlabeled points, noting that not all unlabeled
+        # points at this level have the same parent.
+        unlabeled_mask = cluster_idx_vector == -1
+
+        # At the top level, we don't need to wory about unlabeled points having
+        # different parents.
+        if len(parents[0]):
+            parent_masks = [
+                (parents[0][-1] == parent) for parent in np.unique(parents[0][-1])
+            ]
+        else:
+            parent_masks = [unlabeled_mask]
+
+        # Iterate over possible parents
+        for parent_mask in parent_masks:
+            cluster_mask = unlabeled_mask & parent_mask
+
+            if np.any(cluster_mask):
```

Comment:
```suggestion
            if np.sum(cluster_mask) > 2:
```

I had some errors when this was too small and we ended up with an inability to generate bounds. This seemed to fix it.

Reply:
Yep, good call. I ran into and then promptly forgot about this happening when only plotting a subset of the ArXiV example. This is a nice fix.